src/gwt/src/org/rstudio/studio/client/workbench/BrowseAddinsDialog.java

```diff
@@ -333,17 +342,37 @@ public void execute()
          };
       }
       
-      final String id = selection_.getId();
-      final String encoded = RAddin.encode(selection_);
-      return new Command()
+      return new ExecuteAddinCommand(
+            mruList_,
+            addins_.get(selection_.getId()),
+            RAddin.encode(selection_),
+            new AddinExecutor());
+   }
+   
+   private static class ExecuteAddinCommand implements Command
+   {
+      public ExecuteAddinCommand(AddinsMRUList mruList,
+                                 RAddin addin,
+                                 String encoded,
+                                 AddinExecutor executor)
       {
-         @Override
-         public void execute()
-         {
-            mruList_.add(encoded);
-            server_.executeRAddin(id, new VoidServerRequestCallback());
-         }
-      };
+         mruList_ = mruList;
+         addin_ = addin;
+         encoded_ = encoded;
+         executor_ = executor;
+      }
+      
+      @Override
+      public void execute()
+      {
+         mruList_.add(encoded_);
```

Comment:
The MRUList `add()` method actually does perform a `prepend` (it's the only add operation available and since appending to an MRU usually doesn't make sense that interface isn't available)